scenes/cslib_array.txt

```diff
@@ -247,6 +247,180 @@
 *return
 
 
+*comment TO_STRING
+*comment ------------------------------
+*comment Returns all elements of an array joined as a string.
+*comment An optional string may be passed with which
+*comment to separate the elements.
+*comment ------------------------------
+*comment    params:
+*comment        p_array_ref(string): the name of the array to convert to a string
+*comment        p_separator(string)?: the string to insert between each array element
+*comment    returns:
+*comment        (any): all array elements as a string
+*label to_string
+*params p_array_ref
+*temp p_separator "@{(param_count > 1) ${param_2}|}"
+*temp count {p_array_ref&"_count"}
+*set cslib_ret ""
+*if (count < 1)
+    *return
+*temp n 1
+*label _to_string_loop
+*if (n < count)
+    *set cslib_ret &({p_array_ref&"_${n}"}&p_separator)
+    *set n + 1
+    *goto _to_string_loop
+*set cslib_ret &{p_array_ref&"_${n}"}
+*return
+
+
+*comment REVERSE
+*comment ------------------------------
+*comment Reverses the elements in the array.
+*comment ------------------------------
+*comment    params:
+*comment        p_array_ref(string): the name of the array to reverse
+*label reverse
+*params p_array_ref
+*set cslib_ret ""
+*temp count {p_array_ref&"_count"}
+*if count < 2
+    *return
+*temp s_idx 1
+*temp e_idx count
+*temp store
+*label _reverse_loop
+*if (s_idx < e_idx)
+    *set store {p_array_ref&"_${s_idx}"}
+    *set {p_array_ref&"_${s_idx}"} {p_array_ref&"_${e_idx}"}
+    *set {p_array_ref&"_${e_idx}"} store
```

Comment:
A trick to swap two values. The *gosub swap may be more readable

https://forum.choiceofgames.com/t/new-choicescript-features-implicit-control-flow-gosub-parameters/30042/24

Reply:
It's a neat trick, but I don't think it will improve things here as we have to set the global values (not just the temps).
You'll also trash p_array_ref on a second call to *params.